src/services/WorkspaceArtifactsApi.ts

```diff
@@ -0,0 +1,43 @@
+import { Artifact, ArtifactResponse } from '@/models'
+import { ArtifactFilter } from '@/models/Filters'
+import { BatchProcessor } from '@/services/BatchProcessor'
+import { mapper } from '@/services/Mapper'
+import { WorkspaceApi } from '@/services/WorkspaceApi'
+import { toMap } from '@/utilities'
+
+export interface IWorkspaceArtifactsApi {
+  getArtifact: (id: string) => Promise<Artifact>,
+  getArtifacts: (filter: ArtifactFilter) => Promise<Artifact[]>,
+  getArtifactsCount: (filter: ArtifactFilter) => Promise<number>,
+  deleteArtifact: (id: string) => Promise<void>,
+}
+
+export class WorkspaceArtifactsApi extends WorkspaceApi implements IWorkspaceArtifactsApi {
+
+  protected override routePrefix = '/artifacts'
+
+  private readonly batcher = new BatchProcessor<string, Artifact>(async ids => {
+    const artifacts = await this.getArtifacts({ id: ids })
+    return toMap(artifacts, 'id')
+  }, { maxBatchSize: 200 })
```

Comment:
Love how simple this BatchProcessor is

Reply:
Agreed, whoever designed the api did a nice job 😄